### ISSUE TITLE
at higher scale factors, InvalidOperationError: `round` operation not supported for dtype `decimal[*,8]`

### ISSUE THREAD

**Issue by lostmygithubaccount** [pola-rs/polars#17190]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

you will additionally need `pip install 'ibis-framework[duckdb]', though could also reproduce with just duckdb or likely some hand-crafted Parquet files

```python
import os
import ibis
import polars as pl

os.environ["POLARS_VERBOSE"] = "1"
pl.Config.activate_decimals()

# create data
con = ibis.get_backend()
con.raw_sql("call dbgen(sf=1)")

for table in con.list_tables():
    con.table(table).to_parquet(f"{table}.parquet")

# define q8
from datetime import date

def q8(
    customer,
    lineitem,
    nation,
    orders,
    part,
    region,
    supplier,
    **kwargs,
):
    var1 = "BRAZIL"
    var2 = "AMERICA"
    var3 = "ECONOMY ANODIZED STEEL"
    var4 = date(1995, 1, 1)
    var5 = date(1996, 12, 31)

    n1 = nation.select("n_nationkey", "n_regionkey")
    n2 = nation.select("n_nationkey", "n_name")

    q_final = (
        part.join(lineitem, left_on="p_partkey", right_on="l_partkey")
        .join(supplier, left_on="l_suppkey", right_on="s_suppkey")
        .join(orders, left_on="l_orderkey", right_on="o_orderkey")
        .join(customer, left_on="o_custkey", right_on="c_custkey")
        .join(n1, left_on="c_nationkey", right_on="n_nationkey")
        .join(region, left_on="n_regionkey", right_on="r_regionkey")
        .filter(pl.col("r_name") == var2)
        .join(n2, left_on="s_nationkey", right_on="n_nationkey")
        .filter(pl.col("o_orderdate").is_between(var4, var5))
        .filter(pl.col("p_type") == var3)
        .select(
            pl.col("o_orderdate").dt.year().alias("o_year"),
            (pl.col("l_extendedprice") * (1 - pl.col("l_discount"))).alias("volume"),
            pl.col("n_name").alias("nation"),
        )
        .with_columns(
            pl.when(pl.col("nation") == var1)
            .then(pl.col("volume"))
            .otherwise(0)
            .alias("_tmp")
        )
        .group_by("o_year")
        .agg((pl.sum("_tmp") / pl.sum("volume")).round(2).alias("mkt_share"))
        .sort("o_year")
    )

    return q_final

res = q8(
    customer=pl.scan_parquet("customer.parquet"),
    lineitem=pl.scan_parquet("lineitem.parquet"),
    nation=pl.scan_parquet("nation.parquet"),
    orders=pl.scan_parquet("orders.parquet"),
    part=pl.scan_parquet("part.parquet"),
    region=pl.scan_parquet("region.parquet"),
    supplier=pl.scan_parquet("supplier.parquet"),
)
res.collect()
```

### Log output

```
found multiple sources; run comm_subplan_elim
keys/aggregates are not partitionable: running default HASH AGGREGATION
join parallel: false
join parallel: false
join parallel: false
join parallel: false
join parallel: false
join parallel: false
join parallel: false
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
INNER join dataframes finished
INNER join dataframes finished
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
parquet file must be read, statistics not sufficient for predicate.
INNER join dataframes finished
INNER join dataframes finished
CACHE SET: cache id: 0
INNER join dataframes finished
parquet file must be read, statistics not sufficient for predicate.
INNER join dataframes finished
CACHE HIT: cache id: 0
INNER join dataframes finished
---------------------------------------------------------------------------
InvalidOperationError                     Traceback (most recent call last)
Cell In[9], [line 10](vscode-notebook-cell:?execution_count=9&line=10)
      [1](vscode-notebook-cell:?execution_count=9&line=1) res = q8(
      [2](vscode-notebook-cell:?execution_count=9&line=2)     customer=pl.scan_parquet("customer.parquet"),
      [3](vscode-notebook-cell:?execution_count=9&line=3)     lineitem=pl.scan_parquet("lineitem.parquet"),
   (...)
      [8](vscode-notebook-cell:?execution_count=9&line=8)     supplier=pl.scan_parquet("supplier.parquet"),
      [9](vscode-notebook-cell:?execution_count=9&line=9) )
---> [10](vscode-notebook-cell:?execution_count=9&line=10) res.collect()

File ~/repos/ibis-bench/venv/lib/python3.11/site-packages/polars/lazyframe/frame.py:1967, in LazyFrame.collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, slice_pushdown, comm_subplan_elim, comm_subexpr_elim, cluster_with_columns, no_optimization, streaming, background, _eager, **_kwargs)
   [1964](https://file+.vscode-resource.vscode-cdn.net/Users/cody/repos/ibis-bench/~/repos/ibis-bench/venv/lib/python3.11/site-packages/polars/lazyframe/frame.py:1964) # Only for testing purposes atm.
   [1965](https://file+.vscode-resource.vscode-cdn.net/Users/cody/repos/ibis-bench/~/repos/ibis-bench/venv/lib/python3.11/site-packages/polars/lazyframe/frame.py:1965) callback = _kwargs.get("post_opt_callback")
-> [1967](https://file+.vscode-resource.vscode-cdn.net/Users/cody/repos/ibis-bench/~/repos/ibis-bench/venv/lib/python3.11/site-packages/polars/lazyframe/frame.py:1967) return wrap_df(ldf.collect(callback))

InvalidOperationError: `round` operation not supported for dtype `decimal[*,8]`
```

### Issue description

if you use `sf=0.1` instead of `sf=1`, the query is fine. this is relatively consistent with other issues I noticed with decimals at higher scale factors in my benchmarking, though was surprising here...or it's varying on different runs? it is reproducible though sometimes seems to work

xref: #17189 

### Expected behavior

the above works

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.31
Index type:           UInt32
Platform:             macOS-14.5-arm64-arm-64bit
Python:               3.11.8 (main, Feb 20 2024, 20:00:15) [Clang 14.0.3 (clang-1403.0.22.14.1)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               2024.5.0
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         1.6.0
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               1.5.3
pyarrow:              16.1.0
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
torch:                <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```
</details>

**Comment by JulianCologne** [comment]
I do not think this has anything to do with some special scale factors.

This fails because `round` is not yet implemented for `Decimal` (see my issuel here #15151)

```python
pl.DataFrame({"x": [D("1.23")]}).with_columns(r=pl.col("x").round(decimals=2))
# InvalidOperationError: `round` operation not supported for dtype `decimal[*,2]`
```

**Comment by lostmygithubaccount** [comment]
I'll close this as a duplicate and follow along over there. thanks!

(it does still seem odd it succeeds at the lower scale factor...)
